### DISCUSSION TITLE
Not all endpoints use the proxy

### DISCUSSION THREAD

**Discussion by stargazerZJ** [danny-avila/LibreChat#3462] [Troubleshooting] [answered]
### What happened?

I've installed and configured the latest version of Libre Chat with docker compose, locally on a Linux server. The server is behind a firewall so that every request to the internet must go through a http proxy.

I've configured the PROXY in the `.env` file, as the document requires.

I tested and found that although OpenAI and Anthropic requests uses the proxy correctly, some web access does not use the proxy, and fails, including:
- https://generativelanguage.googleapis.com/v1/models
- https://api.cohere.ai/v1
- OpenAI model fetching also fails
I can confirm that these requests does not go through the proxy through the access logs of the proxy server.

Addition info:
I've tried to debug and found that some proxy-related environment variables are set in every docker container because of the global proxy setting of docker in `~/.docker/config.json`
```bash
sudo docker exec -it LibreChat ash 
/app # env | grep proxy
no_proxy=127.0.0.0/8
all_proxy=http://<address and port>
https_proxy=http://<address and port>
http_proxy=http://<address and port>
/app # env | grep PROXY
HTTPS_PROXY=http://<address and port>
NO_PROXY=127.0.0.0/8
ALL_PROXY=http://<address and port>
HTTP_PROXY=http://<address and port>
```

### Steps to Reproduce

1. Pull the latest repository of Libre Chat
2. Install Libre Chat via docker compose on Linux
3. Set the PROXY in the `.env` file
4. Configure Gemini and Cohere endpoints (the Cohere endpoints is configured in `librechat.yaml`)
5. Use these endpoints.

### Relevant log output

```shell
LibreChat         | 2024-07-27 06:09:41 error: [handleAbortError] AI response error; aborting request: [GoogleGenerativeAI Error]: Error fetching from https://generativelanguage.googleapis.com/v1/models/gemini-pro:streamGenerateContent?alt=sse: fetch failed

LibreChat         | 2024-07-27 06:08:32 error: Failed to fetch models from OpenAI API
LibreChat         | The request either timed out or was unsuccessful. Error message:
LibreChat         |  Cannot read properties of undefined (reading 'status')
LibreChat         | 2024-07-27 06:08:33 error: Failed to fetch models from OpenAI API The request was made and the server responded with a status code that falls out of the range of 2xx: Request failed with status code 400. Error response data:
LibreChat         | 
LibreChat         | 2024-07-27 06:08:33 error: Failed to fetch models from OpenAI API
LibreChat         | The request either timed out or was unsuccessful. Error message:
LibreChat         |  Cannot read properties of undefined (reading 'status')
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
not all endpoints support proxying services, namely gemini and cohere

**Reply by stargazerZJ** [reply]
Oh fine. Can you show me some workarounds? I have tried to set up reverse proxy. However, the gemini client still connects to generativelanguage.googleapis.com and fails, just as before, despite the `GOOGLE_REVERSE_PROXY` being set in `.env`. The Cohere client, after changing the base URL, does uses the proxy as well as the reverse proxy, but got a 405 response code error.
